x-pack/plugins/fleet/server/services/actions/types.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  createAction,
+  bulkCreateActions,
+  getActionsByIds,
+  getActionResultsWithKuery,
+  getActionResultsByIds,
+  getActionsWithKuery,
+} from './actions';
+
+export interface FleetActionsClientInterface {
+  /**
+   *
+   * @param action
+   * @returns {id: string; created_at: string}
+   * @throws {FleetActionsError}
+   * creates a new action document
+   */
+  createAction(action: FleetAction): Promise<ReturnType<typeof createAction>>;
+
+  /**
+   *
+   * @param actions
+   * @returns {id: string; created_at: string}[]
+   * @throws {FleetActionsError}
+   * creates multiple action documents
+   * return successfully created actions
+   * logs failed actions documents
+   */
+  bulkCreateActions(actions: FleetAction[]): Promise<ReturnType<typeof bulkCreateActions>>;
+
+  /**
+   *
+   * @param actionIds
+   * @returns {FleetAction[]}
+   * @throws {FleetActionsError}
+   * returns actions by action ids
+   */
+  getActionsByIds(actionIds: string[]): Promise<ReturnType<typeof getActionsByIds>>;
```

Comment:
@paul-tavares For both actions and action results it made sense to be able to query with multiple action ids. Let me know what you think.